app/identifiers/register/establishers/company/IsCompanyDormantId.scala

```diff
@@ -18,13 +18,53 @@ package identifiers.register.establishers.company
 
 import identifiers.TypedIdentifier
 import identifiers.register.establishers.EstablishersId
+import models.Link
 import models.register.DeclarationDormant
+import play.api.i18n.Messages
 import play.api.libs.json.JsPath
+import utils.UserAnswers
+import utils.checkyouranswers.{CheckYourAnswers, CheckYourAnswersCompany}
+import viewmodels.AnswerRow
 
 case class IsCompanyDormantId(index: Int) extends TypedIdentifier[DeclarationDormant] {
   override def path: JsPath = EstablishersId(index).path \ IsCompanyDormantId.toString
 }
 
 object IsCompanyDormantId {
   override def toString: String = "isCompanyDormant"
+
+  implicit def cya(implicit userAnswers: UserAnswers, messages: Messages): CheckYourAnswers[IsCompanyDormantId] = {
+    new CheckYourAnswersCompany[IsCompanyDormantId] {
+
+      private def label(index: Int, ua: UserAnswers): String =
+        dynamicMessage(index, ua, "messages__company__cya__dormant")
+
+      private def hiddenLabel(index: Int, ua: UserAnswers): String =
+        dynamicMessage(index, ua, "messages__visuallyhidden__dynamic_company__dormant")
+
+      override def row(id: IsCompanyDormantId)(changeUrl: String, userAnswers: UserAnswers): Seq[AnswerRow] =
+        userAnswers.get(id) match {
+          case Some(DeclarationDormant.Yes) =>
+            Seq(
+              AnswerRow(
```

Comment:
can we not use StringCYA for this?

Reply:
No can't use stringcya because isdormant id is enumerated type not string. Get compile errors.